File: ui/types/vod.ts
export type VOD = {
    id: string;
    livestreamId: string;
    userId: string;
    title: string;
    description: string | null;
    thumbnailUrl: string | null;
    visibility: "public" | "private";
    viewCount: number;
    duration: number;
    playbackUrl: string;
    createdAt: string; // ISO 8601 timestamp
    updatedAt: string; // ISO 8601 timestamp
};

File: backend/livestream/gateway/vod/vod.go
package vod

import (
	"context"

	"github.com/gofrs/uuid/v5"
)

type CreateVODRequest struct {
	LivestreamId string `json:"livestreamId"`
	UserId       string `json:"userId"`
	Title        string `json:"title"`
	Description  string `json:"description,omitempty"`
	ThumbnailURL string `json:"thumbnailUrl,omitempty"`
	PlaybackURL  string `json:"playbackUrl,omitempty"`
	Duration     int64  `json:"duration"`
}

type VODGateway interface {
	CreateVOD(ctx context.Context, req CreateVODRequest) (*uuid.UUID, error)
}
